### PR TITLE
refactor(src): ensure proper testing and usage of deprecated items

### PR DESCRIPTION
**Pull request by CITguy**
JIRA: https://jira.rax.io/browse/FRMW-1252

### LGTMs
- [x] Dev LGTM
- [x] Dev+Vidyo LGTM

### CODE REVIEW

File: src/elements/Tables/scripts/rxPaginate.js [hunk 1]
```diff
@@ -511,7 +511,7 @@ angular.module('encore.ui.elements')
  * @param {String=} errorMessage
  * An error message that should be displayed if a call to the request fails
  */
-.directive('rxPaginate', function ($q, $compile, debounce, PageTracking, rxPromiseNotifications) {
+.directive('rxPaginate', function ($q, $compile, debounce, rxPromiseNotifications) {
```
Comment (cathysiller) [inline src/elements/Tables/scripts/rxPaginate.js]: is rxPageTracker supposed to be in this function or was PageTracking in original version unnecessary?

Comment (CITguy) [inline src/elements/Tables/scripts/rxPaginate.js]: I discovered that it wasn't being used.

File: src/components/rxOptionTable/scripts/rxOptionTable.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 angular.module('encore.ui.rxOptionTable')
 /**
+ * @deprecated This directive will be removed in EncoreUI 4.0.0
```
Comment (CITguy) [inline src/components/rxOptionTable/scripts/rxOptionTable.js]: It's marked as deprecated in the demo. Making sure it's marked in the ngdocs, too.

File: src/utilities/rxApply/scripts/rxApply.js [hunk 3]
```diff
@@ -32,11 +32,13 @@
...
-        );
-        return $filter('rxApply');
+    function ApplyFilter () {
+        return function (list, filter) {
+            console.warn(
+                'DEPRECATED: Apply - Please use rxApply. ' +
+                'Apply will be removed in EncoreUI 4.0.0'
+            );
+            return rxApplyFilter()(list, filter);
+        };
```
Comment (CITguy) [inline src/utilities/rxApply/scripts/rxApply.js]: This change ensures that the deprecation warning only displays when the `Apply` filter is used.

**Comment by cathysiller** [comment]
Dev LGTM

File: src/utilities/rxEnvironment/scripts/rxEnvironment.spec.js [hunk 4]
```diff
@@ -150,3 +150,156 @@ describe('rxEnvironment', function () {
         expect(envSvc.get().name).to.equal('custom');
     });
 });
+
+describe('service:Environment (DEPRECATED)', function () {
```
Comment (CITguy) [inline src/utilities/rxEnvironment/scripts/rxEnvironment.spec.js]: Make sure legacy/alias still works the same.

File: src/utilities/rxEnvironmentMatch/docs/examples/rxEnvironmentMatch.simple.html [hunk 5]
```diff
@@ -1,8 +1,3 @@
...
-  <p>
-    This example checks if you are in a <code>ghPages</code> environment using <code>rxEnvironmentMatch</code> filter.
-  </p>
-  <p>
-    Are you in the <code>ghPages</code> environment? <b>{{ 'ghPages' | rxEnvironmentMatch }}</b>
-  </p>
-</div>
+<p>
+  Are you in the <code>ghPages</code> environment? <b>{{ 'ghPages' | rxEnvironmentMatch }}</b>
+</p>
```
Comment (CITguy) [inline src/utilities/rxEnvironmentMatch/docs/examples/rxEnvironmentMatch.simple.html]: Simplified example. 

* Controller unnecessary
* Extra copy unnecessary

File: src/utilities/rxEnvironmentMatch/docs/examples/rxEnvironmentMatch.simple.js [hunk 6]
```diff
@@ -1,4 +0,0 @@
-angular.module('demoApp')
-.controller('rxEnvironmentMatchSimpleCtrl', function ($scope, Environment) {
-    $scope.Environment = Environment;
```
Comment (CITguy) [inline src/utilities/rxEnvironmentMatch/docs/examples/rxEnvironmentMatch.simple.js]: Example never used `Environment`, so the controller is being removed to simplify the example.

File: src/utilities/rxEnvironmentUrl/docs/examples/rxEnvironmentUrl.simple.js [hunk 7]
```diff
@@ -1,4 +0,0 @@
-angular.module('demoApp')
-.controller('rxEnvironmentUrlSimpleCtrl', function ($scope, rxEnvironment) {
-    $scope.rxEnvironment = rxEnvironment;
```
Comment (CITguy) [inline src/utilities/rxEnvironmentUrl/docs/examples/rxEnvironmentUrl.simple.js]: Markup never uses `rxEnvironment`, so the controller is being removed to simplify the example.

File: src/utilities/rxErrorFormatter/scripts/rxErrorFormatter.spec.js [hunk 8]
```diff
@@ -33,3 +33,39 @@ describe('utilities: rxErrorFormatter', function () {
         expect(errorMsg).to.equal('foo bar');
     });
 });
+
+describe('service:ErrorFormatter (DEPRECATED)', function () {
```
Comment (CITguy) [inline src/utilities/rxErrorFormatter/scripts/rxErrorFormatter.spec.js]: Ensure `ErrorFormatter` alias still works as it currently does.

File: src/utilities/rxPageTracker/scripts/rxPageTracker.spec.js [hunk 9]
```diff
@@ -1,322 +1,641 @@
-describe('Pagination', function () {
-    describe('Factory: rxPageTracker', function () {
```
Comment (CITguy) [inline src/utilities/rxPageTracker/scripts/rxPageTracker.spec.js]: Removed unnecessary `describe()`

File: src/utilities/rxPager/scripts/rxPager.spec.js [hunk 10]
```diff
@@ -1,56 +1,109 @@
-describe('Pagination', function () {
-    describe('Filter: rxPager', function () {
```
Comment (CITguy) [inline src/utilities/rxPager/scripts/rxPager.spec.js]: Remove unnecessary `describe()` block.

File: src/utilities/rxPager/scripts/rxPager.spec.js [hunk 11]
```diff
@@ -1,56 +1,109 @@
...
-            // Test last page that has even # of custom pagesToShow and is below pagesToShowLimit
-            expect(page({ pageNumber: 4, totalPages: 4, pagesToShow: 4 })).to.eql([0,1,2,3]);
+        // Test last page that has even # of custom pagesToShow and is above pagesToShowLimit
+        expect(page({ pageNumber: 102, totalPages: 102, pagesToShow: 4 })).to.eql([98,99,100,101]);
+    });
+});
 
-            // Test first page that has even # of custom pagesToShow and is above pagesToShowLimit
-            expect(page({ pageNumber: 1, totalPages: 102, pagesToShow: 4 })).to.eql([0,1,2,3]);
+describe('filter:Page (DEPRECATED)', function () {
```
Comment (CITguy) [inline src/utilities/rxPager/scripts/rxPager.spec.js]: Ensure `Page` still works as it does today.

File: src/utilities/rxSelectFilter/scripts/rxSelectFilter.spec.js [hunk 12]
```diff
@@ -58,3 +58,64 @@ describe('utilities:rxSelectFilter', function () {
         });
     });
 });
+
+describe('service:SelectFilter (DEPRECATED)', function () {
```
Comment (CITguy) [inline src/utilities/rxSelectFilter/scripts/rxSelectFilter.spec.js]: Ensure `SelectFilter` still works as it does today.

File: src/utilities/rxSessionStorage/scripts/rxSessionStorage.spec.js [hunk 13]
```diff
@@ -57,3 +57,63 @@ describe('service:rxSessionStorage', function () {
         expect(storage.length).to.eq(0);
     });
 });
+
+describe('service:SessionStorage (DEPRECATED)', function () {
```
Comment (CITguy) [inline src/utilities/rxSessionStorage/scripts/rxSessionStorage.spec.js]: Ensure `SessionStorage` still works as it does today.

File: src/utilities/rxStatus/scripts/rxStatus.spec.js [hunk 14]
```diff
@@ -1,5 +1,4 @@
-// TODO: refactor tests
-describe('utilities: rxStatus', function () {
+describe('service:rxStatus', function () {
```
Comment (CITguy) [inline src/utilities/rxStatus/scripts/rxStatus.spec.js]: File was refactored to make better use of `beforeEach()` in appropriate `describe()` blocks.

File: src/utilities/rxStatus/scripts/rxStatus.spec.js [hunk 15]
```diff
@@ -20,79 +19,331 @@ describe('utilities: rxStatus', function () {
...
+                var spy = sinon.spy(rxStatus, 'setStack');
+                rootScope.$broadcast('$routeChangeStart');
+                expect(rxStatus.setStack.args[0][0]).to.equal('page');
+                spy.restore();
+            });
+        });
+    });//on route load
+});//rxStatus
+
+describe('service:Status (DEPRECATED)', function () {
```
Comment (CITguy) [inline src/utilities/rxStatus/scripts/rxStatus.spec.js]: Ensure `Status` still works as it does today.

File: src/utilities/rxTitleize/scripts/rxTitleize.spec.js [hunk 16]
```diff
@@ -18,7 +18,7 @@ describe('rxTitleize', function () {
     });
 });
 
-describe('titleize (DEPRECATED)', function () {
+describe('filter:titleize (DEPRECATED)', function () {
```
Comment (CITguy) [inline src/utilities/rxTitleize/scripts/rxTitleize.spec.js]: Ensure `titleize` still works as it does today.

File: src/utilities/rxTokenInterceptor/scripts/rxTokenInterceptor.spec.js [hunk 17]
```diff
@@ -60,8 +60,7 @@ describe('service:rxTokenInterceptor', function () {
     });
 });
 
-// TODO: Deprecated! Remove for 4.0.0
-describe('service:TokenInterceptor', function () {
+describe('provider:TokenInterceptor (DEPRECATED)', function () {
```
Comment (CITguy) [inline src/utilities/rxTokenInterceptor/scripts/rxTokenInterceptor.spec.js]: Ensure `TokenInterceptor` still works as it does today.

File: src/utilities/rxUnauthorizedInterceptor/scripts/rxUnauthorizedInterceptor.spec.js [hunk 18]
```diff
@@ -76,3 +76,82 @@ describe('service:rxUnauthorizedInterceptor', function () {
         expect(mockWindow.location).to.contain('/mario');
     });
 });
+
+describe('service:UnauthorizedInterceptor (DEPRECATED)', function () {
```
Comment (CITguy) [inline src/utilities/rxUnauthorizedInterceptor/scripts/rxUnauthorizedInterceptor.spec.js]: Ensure `UnauthorizedInterceptor` alias still works as it does today.

File: src/utilities/rxUrlUtils/docs/rxUrlUtils.meta.json [hunk 19]
```diff
@@ -1,7 +1,9 @@
...
-    "displayName": "urlUtils",
+    "displayName": "rxUrlUtils",
     "stability": "stable",
     "description": "Set of utility functions to break apart/compare URLs.",
-    "api": "service:urlUtils",
-    "keywords": []
+    "api": "service:rxUrlUtils",
+    "keywords": [
+        "urlUtils"
+    ]
```
Comment (CITguy) [inline src/utilities/rxUrlUtils/docs/rxUrlUtils.meta.json]: Updated to reflect change in service name/alias.

File: src/utilities/rxUrlUtils/scripts/rxUrlUtils.spec.js [hunk 20]
```diff
@@ -223,7 +223,7 @@ describe('rxUrlUtils', function () {
     });
 });
 
-describe('urlUtils (DEPRECATED)', function () {
+describe('service:urlUtils (DEPRECATED)', function () {
```
Comment (CITguy) [inline src/utilities/rxUrlUtils/scripts/rxUrlUtils.spec.js]: Ensure `urlUtils` alias still works as it does today.

File: src/utilities/rxXor/scripts/rxXor.js [hunk 21]
```diff
@@ -27,11 +27,13 @@
...
-        );
-        return $filter('rxXor');
+    function xorFilter () {
+        return function (a, b) {
+            console.warn(
+                'DEPRECATED: xor - Please use rxXor. ' +
+                'xor will be removed in EncoreUI 4.0.0'
+            );
+            return rxXorFilter()(a, b);
+        };
```
Comment (CITguy) [inline src/utilities/rxXor/scripts/rxXor.js]: This change ensures that the deprecation warning only displays when the `xor` filter is _used_.

File: src/utilities/rxPager/scripts/rxPager.js [hunk 22]
```diff
@@ -45,19 +45,21 @@
...
-        );
-        return $filter('rxPager');
+    function PageFilter () {
+        return function (pager) {
+            console.warn(
+                'DEPRECATED: Page - Please use rxPager. ' +
+                'Page will be removed in EncoreUI 4.0.0'
+            );
+            return rxPagerFilter()(pager);
+        };
```
Comment (CITguy) [inline src/utilities/rxPager/scripts/rxPager.js]: This change ensures that the deprecation warning only displays when the `Page` filter is _used_.

**Comment by halkon** [comment]
Vidyo LGTM

**Comment by CITguy** [comment]
Removed "QE LGTM", as per @Droogans states that the big green check mark (tests are passing) is the majority of the review.

**Pull request merged**